src/NuGet.Core/NuGet.Packaging/Signing/TrustStore/FallbackCertificateBundleX509ChainFactory.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if NET5_0_OR_GREATER
+
+using System;
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+
+namespace NuGet.Packaging.Signing
+{
+    internal sealed class FallbackCertificateBundleX509ChainFactory : CertificateBundleX509ChainFactory
+    {
+        internal const string FileName = "cstsroots.pem";
+
+        private static readonly Lazy<string> ThisAssemblyDirectoryPath = new(GetThisAssemblyDirectoryPath, LazyThreadSafetyMode.ExecutionAndPublication);
+
+        private FallbackCertificateBundleX509ChainFactory(X509Certificate2Collection certificates, string filePath)
+            : base(certificates, filePath)
+        {
+        }
+
+        internal static bool TryCreate(out FallbackCertificateBundleX509ChainFactory factory, string filePath = FileName)
+        {
+            factory = null;
+
+            string fullFilePath = Path.IsPathRooted(filePath) ? filePath : Path.Combine(ThisAssemblyDirectoryPath.Value, filePath);
+
+            if (TryImportFromPemFile(fullFilePath, out X509Certificate2Collection certificates))
+            {
+                factory = new FallbackCertificateBundleX509ChainFactory(certificates, fullFilePath);
+
+                return true;
+            }
+
+            return false;
+        }
+
+        private static string GetThisAssemblyDirectoryPath()
+        {
+            string location = typeof(FallbackCertificateBundleX509ChainFactory).Assembly.Location;
+            FileInfo thisAssembly = new(location);
```

Comment:
nit: This logic is pretty simple, maybe you don't need an entire method and just specify the logic in the lamda?

```c#
private static readonly Lazy<string> ThisAssemblyDirectoryPath = new(() => Path.GetDirectoryName(typeof(FallbackCertificateBundleX509ChainFactory).Assembly.Location)), LazyThreadSafetyMode.ExecutionAndPublication);
```

Reply:
I think a well-named method improves readability over a longer statement with a bunch of nested expressions.  No harm is there?